stan/math/gpu/opencl_context.hpp

```diff
@@ -229,7 +239,7 @@ class opencl_context {
 
       try {
         std::vector<cl::Device> all_devices;
-        platform.getDevices(CL_DEVICE_TYPE_GPU, &all_devices);
+        platform.getDevices(CL_DEVICE_TYPE_ALL, &all_devices);
```

Comment:
❤️

Reply:
Going to add here that I need to go fix the wiki so that users can check
```
clinfo -l
```

to get the device types / platforms and their numbers to pick.

Another interesting bit of this, moving over to OpenCL 2.0 and using shared virtual memory with the device being CPU would let us pay zero copy cost and still get parallelization.